service/switchboardapp/src/test/java/com/switchboard/app/EncoderControllerTest.java

```diff
@@ -0,0 +1,112 @@
+package com.switchboard.app;
+
+import com.switchboard.app.controller.EncoderController;
+import com.switchboard.app.dao.DeviceDaoImpl;
+import com.switchboard.app.dao.EncoderDaoImpl;
+import com.switchboard.app.domain.DeviceEntity;
+import com.switchboard.app.domain.EncoderEntity;
+import com.switchboard.app.exceptions.DeviceNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.ResponseEntity;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+class EncoderControllerTest {
+    @InjectMocks
+    EncoderController encoderController;
+
+    @Mock
+    EncoderDaoImpl encoderService;
+
+    @Mock
+    DeviceDaoImpl deviceService;
+
+    //stubbed Objects
+    private DeviceEntity device1;
+    private EncoderEntity encoder1;
+
+    @BeforeEach
+    void setUp(){
```

Comment:
Typo here, `setup`

Reply:
@FGRCL Where is the typo? setUp() is a junit method that he is overriding.

Reply:
I've changed it to setup